### PR TITLE
[trel] disable trel if feature flag is used

### PR DESCRIPTION
**Pull request by sunytt**
This PR sets trel to disabled if feature flag is used.

Added a test in border_router.yml with both trel and feature flag compiled.

Updated border_router.yml tests to use the OTBR_TREL flag.

### CODE REVIEW

File: .github/workflows/border_router.yml [hunk 1]
```diff
@@ -50,54 +50,61 @@ jobs:
       matrix:
         include:
           - name: "Border Router (mDNSResponder)"
-            otbr_options: "-DOT_TREL=OFF -DOT_MLR=ON -DOTBR_COVERAGE=ON -DOT_SRP_SERVER=ON -DOT_ECDSA=ON -DOT_SERVICE=ON -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON"
+            otbr_options: "-DOTBR_TREL=OFF -DOT_MLR=ON -DOTBR_COVERAGE=ON -DOT_SRP_SERVER=ON -DOT_ECDSA=ON -DOT_SERVICE=ON -DOTBR_DUA_ROUTING=ON -DOT_DUA=ON"
```
Comment (superwhd) [inline .github/workflows/border_router.yml:53]: Shall we group OTBR options and OT options respectively?

File: src/ncp/ncp_openthread.cpp [hunk 2]
```diff
@@ -215,6 +215,11 @@ void ControllerOpenThread::Init(void)
         VerifyOrExit(result == OT_ERROR_NONE, error = OTBR_ERROR_OPENTHREAD);
     }
 
+#if OTBR_ENABLE_FEATURE_FLAGS && OTBR_ENABLE_TREL
+    // Disable trel when feature flag is enabled, let feature flag handle enable/disable trel
+    otTrelSetEnabled(mInstance, false);
+#endif
```
[the remark this answers is not in the document]
Comment (sunytt) [reply src/ncp/ncp_openthread.cpp:221]: Yes, by default trel is enabled if feature_flags is disabled.
This is to keep the default behavior so if other teams rely on it then we won't break their code.

**Review by superwhd — APPROVED** [review]
LGTM :+1:

**Review by abtink — APPROVED** [review]
👍

File: src/ncp/ncp_openthread.cpp [hunk 3]
```diff
@@ -215,6 +215,11 @@ void ControllerOpenThread::Init(void)
         VerifyOrExit(result == OT_ERROR_NONE, error = OTBR_ERROR_OPENTHREAD);
     }
 
+#if OTBR_ENABLE_FEATURE_FLAGS && OTBR_ENABLE_TREL
+    // Disable trel when feature flag is enabled, let feature flag handle enable/disable trel
+    otTrelSetEnabled(mInstance, false);
```
Comment (abtink) [inline src/ncp/ncp_openthread.cpp:220]: Is it possible to get here with feature flags wanting the TREL to be enabled already?

Comment (sunytt) [reply src/ncp/ncp_openthread.cpp:220]: Good point, the feature flags are updated by thread daemon after otbr-agent signals OTBR_DBUS_SIGNAL_READY, so here we don't get the updated value. But the default feature flag value can be 'true' or 'false' so I think we should use that instead. Will update.

**Review by jwhui — APPROVED** [review]

**Pull request merged**